src/main/java/pl/javastart/couponcalc/PriceCalculator.java

```diff
@@ -1,13 +1,73 @@
 package pl.javastart.couponcalc;
 
+import java.math.BigDecimal;
+import java.math.RoundingMode;
 import java.util.List;
 
 public class PriceCalculator {
 
+    private static final int ONE_COUPON = 1;
+
     public double calculatePrice(List<Product> products, List<Coupon> coupons) {
-        // TODO
+        double sumOfPrices = 0;
+        double theMostFavorablePrice = Double.MAX_VALUE;
+        if (products == null || products.isEmpty()) {
+            return 0;
+        } else if (coupons == null) {
+            return getSumWithoutCoupon(products);
+        } else if (coupons.size() >= ONE_COUPON) {
+            for (int i = 0; i < coupons.size(); i++) {
```

Comment:
Dodaj w tym pierwszym else if warunek, czy lista jest pusta, dalej else nie będzie potrzebny, bo co najmniej 1 kupon to jedyna możliwość, nie ma potrzeby sprawdzania `>= ONE_COUPON`